x-pack/plugins/fleet/server/services/epm/archive/extract.ts

```diff
@@ -16,19 +18,20 @@ export async function untarBuffer(
   buffer: Buffer,
   filter = (entry: ArchiveEntry): boolean => true,
   onEntry = (entry: ArchiveEntry): void => {}
-): Promise<unknown> {
+) {
   const deflatedStream = bufferToStream(buffer);
   // use tar.list vs .extract to avoid writing to disk
-  const inflateStream = tar.list().on('entry', (entry: tar.FileStat) => {
-    const path = entry.header.path || '';
+  const inflateStream = tar.list().on('entry', (entry) => {
+    const path = entry.path || '';
     if (!filter({ path })) return;
-    streamToBuffer(entry).then((entryBuffer) => onEntry({ buffer: entryBuffer, path }));
+    streamToBuffer(entry as unknown as NodeJS.ReadableStream).then((entryBuffer) =>
+      onEntry({ buffer: entryBuffer, path })
+    );
   });
 
-  return new Promise((resolve, reject) => {
-    inflateStream.on('end', resolve).on('error', reject);
-    deflatedStream.pipe(inflateStream);
-  });
+  deflatedStream.pipe(inflateStream);
+
+  await finished(inflateStream);
```

Comment:
According to the types `inflateStream` will never emit an `error` event. I'm not sure if we can 100% trust this, but using `finished` instead of listening for `end` is normally better practice anyway, and by doing so we actually also get any `error` events that might occur (without having to work around TS telling us it's not happening).